v1.4 PED fix & Update of PPP [Complete]

DonovanJJ opened the pull request:
## The following PED bug fixes were made:
- 91, 93, 122 (Removed the existing add command)
- 127, 128, 131 (Convert multiple white spaces into single white spaces and trimmed the user input)
- 104 (UG `remove /c` command was updated)
- 132 (Add checking of attendance boundary when marking student present)
- 106 (Update the error message for remove such that the correct  message is shown, previously was a bug)

## The following bug was fixed (Non-PED):
- 81 (Cascaded the removal of class to students as well)

DonovanJJ committed:
Fix UG bugs in issue 104

DonovanJJ committed:
Fix UG bugs in issue 104

DonovanJJ committed:
Fix issue 132.
Introduced checking of existing attendance against class total attendance

DonovanJJ committed:
Fixed general parser bug to accpet multiple starting/trailing white spaces

DonovanJJ committed:
Removed the unused Add command. One more testcase left to edit

DonovanJJ committed:
Fix issue 81, non-ped bug
Fix bug where students are not removed from a class when a class is deleted

DonovanJJ committed:
Fix bug in issue 106
Changed the error message for a git status
the remove class command when there is parse exception when obtaining index value

DonovanJJ committed:
Fix issue 84
Add check in marstudentabsent to ensure it does not go below 0

DonovanJJ committed:
Update Donovan-PPP

DonovanJJ committed:
Fix checkstyle issues

DonovanJJ committed:
Fix indentation error

DonovanJJ committed:
Merge with upstream master

DonovanJJ committed:
FIx checkstyle error

DonovanJJ committed:
Removed unused imports

DonovanJJ committed:
Fix indentation error

DonovanJJ committed:
Add test case to MarkStudentPresentCommand

DonovanJJ committed:
Merge with upstream PR

songgthu commented on docs/UserGuide.md:
```diff
@@ -871,10 +870,10 @@ There is no need to save manually.
+| **remove**      | **Removing a class:** <br> **Format:** `remove /c CLASS_INDEX` <br> **E.g:** `remove /c 3` <br> <br> **Removing a student:** <br> **Format:** `remove /s STUDENT_INDEX /c CLASS_NAME` <br> **E.g:** `remove /s 1 /c CS2103T`                                                                                                                            |
 | **view**        | **Viewing a class:** <br> **Format:** `view /c CLASS_INDEX` <br> **E.g:** `view /c 1`                                                                                                                                                                                                                                                                    |
-| **edit**        | **Editing a class:** <br> **Format:** `edit /c CLASS_INDEX [/n CLASS_NAME] [/m MEMO] [/t SCHEDULE]` <br> **E.g:** `edit /c 2 /m submit marking report` <br><br> **Editing a student:** <br> **Format:** `edit /s STUDENT_INDEX /c CLASS_NAME [/n STUDENT_NAME] [/id ID] [/m MEMO] [/p PARTICIPATION]` <br> **E.g:** `edit /s 1 /c cs2103t /id A1234567W` |
+| **edit**        | **Editing a class:** <br> **Format:** `edit /c CLASS_INDEX [/n CLASS_NAME] [/m MEMO] [/t SCHEDULE]` <br> **E.g:** `edit /c 2 /m submit marking report` <br><br> **Editing a student:** <br> **Format:** `edit /s STUDENT_INDEX /c CLASS_NAM  E [/n STUDENT_NAME] [/id ID] [/m MEMO] [/p PARTICIPATION]` <br> **E.g:** `edit /s 1 /c cs2103t /id A1234567W` |
```
just a small typo here, there's an extra space in "CLASS_NAME"

DonovanJJ replied:
Thanks for spotting this mistake, I've made the necessary changes to this and the one you pointed out in the command summary!

songgthu commented on docs/UserGuide.md:
```diff
@@ -871,10 +870,10 @@ There is no need to save manually.
 |-----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| **add**         | **Adding a class:** <br> **Format:** `add /c CLASS_NAME` <br> **E.g:** `add /c CS2103T` <br> <br> **Adding a student:** <br> **Format:** `add /s STUDENT_NAME /c CLASS_INDEX` <br> **E.g:** `add /s John /c CS2103T` <br><br> **Adding a list of students:** <br> Coming soon                                                                            |
-| **remove**      | **Removing a class:** <br> **Format:** `remove /c CLASS_INDEX` <br> **E.g:** `remove /c 1` <br> <br> **Removing a student:** <br> **Format:** `remove /s STUDENT_INDEX  /c CLASS_NAME` <br> **E.g:** `remove /s 1 /c CS2103T`                                                                                                                            |
+| **add**         | **Adding a class:** <br> **Format:** `add /c CLASS_NAME` <br> **E.g:** `add /c CS2103T` <br> <br> **Adding a student:** <br> **Format:** `add /s STUDENT_NAME  /c CLASS_NAME` <br> **E.g:** `add /s John /c CS2103T` <br><br> **Adding a list of students:** <br> Coming soon                                                                            |
```
I think for the adding a student the format is add /s STUDENT_NAME /c CLASS_INDEX, can you update it and the following example also thanks!

songgthu reviewed (approved):
LGTM! There are some small details to be updated.

sudarshan2401 commented on src/test/java/seedu/address/logic/LogicManagerTest.java:
```diff
@@ -145,41 +136,41 @@ private void assertCommandFailure(String inputCommand, Class<? extends Throwable
+    //        ModelManager expectedModel = new ModelManager();
+    //        expectedModel.addStudent(expectedPerson);
+    //        assertCommandFailure(addCommand, CommandException.class, expectedMessage, expectedModel);
+    //    }
```
Is this part of the test not going to be used? If so maybe can remove

DonovanJJ replied:
Thanks for pointing it out! I've removed this test

sudarshan2401 reviewed (approved):
LGTM! Thanks for the bug fixes!

DonovanJJ committed:
Merge remote-tracking branch 'upstream/master' into v1.4-fix

DonovanJJ committed:
Fix bugs in Ug
In particular typo in command summary and updated the add student command while also removing unused test cases
